$OpenBSD: patch-lib_awful_client_lua_in,v 1.3 2012/07/09 09:01:54 dcoppa Exp $

Revert commit c0dffca646c1a02276da510bf3587fb7fad8e0e9: "Focus
history: Don't ignore unfocusable clients". This caused focus
problems with some panels, most notably fbpanel and gnome-panel
(upstream git commit c084eb5b085287e4fb2661e834197cf8cf233215)

--- lib/awful/client.lua.in.orig	Mon Jun 11 13:57:35 2012
+++ lib/awful/client.lua.in	Mon Jul  9 10:49:18 2012
@@ -131,10 +131,12 @@ end
 --- Update client focus history.
 -- @param c The client that has been focused.
 function focus.history.add(c)
-    -- Remove the client if its in stack
-    focus.history.delete(c)
-    -- Record the client has latest focused
-    table.insert(data.focus, 1, c)
+    if focus.filter(c) then
+        -- Remove the client if its in stack
+        focus.history.delete(c)
+        -- Record the client has latest focused
+        table.insert(data.focus, 1, c)
+    end
 end
 
 --- Get the latest focused client for a screen in history.
